app/controllers/sudo_mode_handler.rb

```diff
@@ -0,0 +1,218 @@
+# frozen_string_literal: true
+
+class SudoModeHandler
+  # The default rank for authentication factors (lower is better)
+  # - 0 is deliberately left available so we can use it for the user's preference
+  # - `:backup_code` is explicitly omitted as we don't want to use that for
+  #   reauthentication
+  FACTOR_PREFERENCES = {
+    totp: 1,
+    webauthn: 2,
```

Comment:
Should we switch the order of `webauthn` and `totp`? I wonder if WebAuthn is more secure espeically since it supports hardware security keys and biometrics.